### PR TITLE
doc: TransactionCallbacks class documentation

### PR DESCRIPTION
**Pull request by m-blaha**
Add missing documentation to rpm transaction callbacks.

Resolves: https://github.com/rpm-software-management/dnf5/issues/1658

### CODE REVIEW

File: include/libdnf5/rpm/transaction_callbacks.hpp [hunk 1]
```diff
@@ -74,27 +86,92 @@ class LIBDNF_API TransactionCallbacks {
...
+
+    /// Execution of the rpm scriptlet finished with error
+    /// @param item The TransactionPackage class instance for the package that triggered the scriptlet execution.
+    /// @param nevra Nevra of the package that triggered scriptlet execution
+    /// @param type Type of the scriptlet
+    /// @param return_code The return code of the scriptlet execution
     virtual void script_error(const TransactionItem * item, Nevra nevra, ScriptType type, uint64_t return_code);
+    /// Execution of the rpm scriptlet has started
+    /// @param item The TransactionPackage class instance for the package that triggered the scriptlet execution. It can be `nullptr` if the package is not part of the transaction (e.g., a package installation triggered an update of the mandb and man-db was already installed).
+    /// @param nevra Nevra of the package that triggered scriptlet execution
```
[the remark this answers is not in the document]
Comment (m-blaha) [reply include/libdnf5/rpm/transaction_callbacks.hpp:151]: Thank you, I tried to clarify it.

File: include/libdnf5/rpm/transaction_callbacks.hpp [hunk 2]
```diff
@@ -74,27 +86,92 @@ class LIBDNF_API TransactionCallbacks {
...
+    /// @param type Type of the scriptlet
+    /// @param return_code The return code of the scriptlet execution
     virtual void script_error(const TransactionItem * item, Nevra nevra, ScriptType type, uint64_t return_code);
+    /// Execution of the rpm scriptlet has started
+    /// @param item The TransactionPackage class instance for the package that owns the executed or triggered scriptlet. It can be `nullptr` if the scriptlet owner is not part of the transaction (e.g., a package installation triggered an update of the man database, owned by man-db package).
+    /// @param nevra Nevra of the package that owns the executed or triggered scriptlet.
+    /// @param type Type of the scriptlet
     virtual void script_start(const TransactionItem * item, Nevra nevra, ScriptType type);
+    /// Execution of the rpm scriptlet finished without critical error
+    /// @param item The TransactionPackage class instance for the package that owns the executed or triggered scriptlet. It can be `nullptr` if th  e scriptlet owner is not part of the transaction.
```
Comment (jrohel) [inline include/libdnf5/rpm/transaction_callbacks.hpp:155]: `th e` -> `the`

File: include/libdnf5/rpm/transaction_callbacks.hpp [hunk 3]
```diff
@@ -74,27 +86,92 @@ class LIBDNF_API TransactionCallbacks {
...
+    /// Unpacking of the package failed.
+    /// @param item The TransactionPackage class instance representing the package that failed to unpack
     virtual void unpack_error(const TransactionItem & item);
+
+    /// cpio error during the package installation. Currentl not used by librpm.
+    /// @param item The TransactionPackage class instance representing the package that caused the error
     virtual void cpio_error(const TransactionItem & item);
+
+    /// Execution of the rpm scriptlet finished with error
+    /// @param item The TransactionPackage class instance for the package that owns the executed or triggered scriptlet. It can be `nullptr` if th  e scriptlet owner is not part of the transaction.
```
Comment (jrohel) [inline include/libdnf5/rpm/transaction_callbacks.hpp:144]: `th e` -> `the`

File: include/libdnf5/rpm/transaction_callbacks.hpp [hunk 4]
```diff
@@ -74,27 +86,92 @@ class LIBDNF_API TransactionCallbacks {
...
+    /// @param item The TransactionPackage class instance for the package currently being removed
+    /// @param amount The portion of the package already uninstalled
+    /// @param total The disk space freed by the package after removal
     virtual void uninstall_stop(const TransactionItem & item, uint64_t amount, uint64_t total);
+
+    /// Unpacking of the package failed.
+    /// @param item The TransactionPackage class instance representing the package that failed to unpack
     virtual void unpack_error(const TransactionItem & item);
+
+    /// cpio error during the package installation. Currentl not used by librpm.
```
Comment (jrohel) [inline include/libdnf5/rpm/transaction_callbacks.hpp:139]: `Currentl` -> `Currently`

File: include/libdnf5/rpm/transaction_callbacks.hpp (hunk 2 above)
Comment (m-blaha) [reply include/libdnf5/rpm/transaction_callbacks.hpp]: Agh! Sorry for these typos, fixed.

File: include/libdnf5/rpm/transaction_callbacks.hpp (hunk 3 above)
Comment (m-blaha) [reply include/libdnf5/rpm/transaction_callbacks.hpp]: Fixed.

File: include/libdnf5/rpm/transaction_callbacks.hpp (hunk 4 above)
Comment (m-blaha) [reply include/libdnf5/rpm/transaction_callbacks.hpp]: Fixed.

File: include/libdnf5/rpm/transaction_callbacks.hpp [hunk 5]
```diff
@@ -74,27 +86,92 @@ class LIBDNF_API TransactionCallbacks {
...
     /// @param total Number of elements in the rpm transaction
     virtual void before_begin(uint64_t total);
+
     /// Called after the transaction run finished
     /// @param success Whether the rpm transaction was completed successfully
     virtual void after_complete(bool success);
 
+    /// Report the package installation progress periodically.
+    /// @param item The TransactionPackage class instance for the package currently being installed
+    /// @param amount The portion of the package already installed
```
Comment (jrohel) [inline include/libdnf5/rpm/transaction_callbacks.hpp:96]: Please add empty doc line after description. Do this for all methods.
Replace:
```
/// Report the package installation progress periodically.
/// @param item The TransactionPackage class instance for the package currently being installed
/// @param amount The portion of the package already installed

```
with:
```
/// Report the package installation progress periodically.
///
/// @param item The TransactionPackage class instance for the package currently being installed
/// @param amount The portion of the package already installed
```

The blank line is important for proper generation of documentation for Python.

**Review by jrohel — CHANGES_REQUESTED** [review]
Thank you for the PR.
I added one more minor change request in the review.  The change is important in order to properly generate documentation for the Python API.

**Review by jrohel — APPROVED** [review]

**Pull request merged**